### DISCUSSION TITLE
tabAutocompleteModel Not Working as Expected with Continue-dev Integration in VS Code

### DISCUSSION THREAD

**Discussion by latent-variable** [open-webui/open-webui#6720] [Issues]
# Bug Report

---

## Installation Method
```
pip install open-webui
```
## Environment

- **Open WebUI Version:** v0.3.45
- **Ollama (if applicable):** v0.4.0-rc5

- **Operating System:**  Windows 10 , Windows 11 , tested
- **Browser (if applicable):** Chrome, Edge

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [x ] I am on the latest version of both Open WebUI and Ollama.
- [NA] I have included the browser console logs.
- [NA] I have included the Docker container logs.
- [x ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Description

**Bug Summary:**
[Continue-dev tabAutocompleteModel instructions](https://docs.openwebui.com/tutorials/integrations/continue-dev/)
Following the included instructions  results in fil in the middle coding models not performing as expected. Please look at examples

## Reproduction Details

1. **Install open-webui**
   - Run the following command:
     ```bash
     pip install open-webui
     ```

2. **Install continue-dev on VS Code**
   - Open Visual Studio Code.
   - Search for "continue-dev" in the Extensions Marketplace and install it.

3. **Follow tabAutocompleteModel instructions on Continue-dev**
   - Refer to the [tabAutocompleteModel documentation](https://docs.openwebui.com/tutorials/integrations/continue-dev/).
   - Follow each step in the guide to configure tabAutocompleteModel.

### Tested with multiple coding models resulting in wrong behavior

**Screen Recordings:**
**Incorrect behavior:**
```
"tabAutocompleteModel": {
    "title": "Granite Code",
    "provider": "openai",
    "model": "granite-code:latest",
    "useLegacyCompletionsEndpoint": false,
    "apiBase": "http://localhost:443/ollama/v1",
    "apiKey": "open-webui-api"
  }
```
![Untitled video - Made with Clipchamp](https://github.com/user-attachments/assets/997cf927-84ef-44b7-af8d-97a37e9ec547)

**Correct behavior: Basic Ollama setup**
```
"tabAutocompleteModel": {
    "title": "Granite Code",
    "provider": "ollama",
    "model": "granite-code:latest"
  }
```
![Untitled video - Made with Clipchamp (2)](https://github.com/user-attachments/assets/c4c36323-9c31-4a37-a404-91a5a3ca7c07)

**Comment by vills** [comment]
Looks like the problem is in on continue.dev side. Versions 0.8.55 and 0.9.222 works well. Newest versions have broken autocomplete with open-webui->ollama proxy.

**Comment by vills** [comment]
@tjbck looks like the problem with OWUI proxy to ollama instance. I get different results with same request body made directly to Ollama and through OWUI. Continue.dev made changes to handle FIM requests on Ollama side for them since versions i mentioned above. I think OWUI proxy is missing something or converting requests in wrong way.

Response while connecting to Ollama directly:
```
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:15:14.837826Z","response":"\n","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:15:14.860235Z","response":"","done":true,"done_reason":"stop","context":[100003,6894,185,2860,68,25,245,27434,11,245,11691,27434,185,1229,25,245,4107,280,14544,4281,185,1804,25,207,18,13,16,19,16,20,24,185,87,5213,25,1927,100002,69,5005,12,8318,25,207,18,185,66679,12,40543,25,100004,185],"total_duration":9447212750,"load_duration":826933125,"prompt_eval_count":51,"prompt_eval_duration":8590000000,"eval_count":2,"eval_duration":25000000}
```

and the response through OWUI:
```
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.231277128Z","response":" It","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.259607535Z","response":" seems","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.285393442Z","response":" like","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.301851274Z","response":" you","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.317934655Z","response":"'","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.333950778Z","response":"re","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.350060084Z","response":" listing","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.378045467Z","response":" some","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.405762167Z","response":" mn","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.431278346Z","response":"em","done":false}
{"model":"deepseek-coder-v2:16b-lite-instruct-q6_K","created_at":"2024-11-21T12:11:57.450773582Z","response":"onic","done":false}
....
```

Request example (taken from Continue.dev): 
```
{
  "model": "deepseek-coder-v2:16b-lite-instruct-q6_K",

  "prompt": "---\ndoe: a deer, a female deer\nray: a drop of golden sun\npi: 3.14159\nxmas: true",

  "suffix": "french-hens: 3\ncalling-birds:",
  
  "options":
    {
      "temperature": 0.1,
      "top_p": 0.95,
      "num_predict": 600,
      "stop":
        [
          "User:",
          "Assistant:",
          "<...fim...begin...>",
          "<...fim...hole...>",
          "<...fim...end...>",
          "//",
          "<...end...of...sentence...>",
          "/src/",
          "#- coding: utf-8",
          "```"
        ],
      "num_ctx": 8096
    },
  "keep_alive": 1800
}
```

**Comment by SlavikCA** [comment]
I found that this config settings works for me in OpenWeb UI 0.5.3:

https://github.com/open-webui/docs/pull/344/files

**Reply by csaleman** [reply]
@SlavikCA I have been looking for the continue.dev code autocomplete solution for weeks. Thank you!
```

name: Local Assistant
version: 1.0.0
schema: v1
models:
  - name: Qwen2.5 Code 14b
    provider: ollama
    model: qwen2.5-coder:14b
    apiBase: http://YOUROPENWEBUI/ollama
    apiKey: sk-YOUR-API-KEY
    roles:
      - chat
      - edit

  - name: Qwen2.5 Code 1.5b
    provider: ollama
    model: qwen2.5-coder:1.5b
    apiBase: http://YOUROPENWEBUI/ollama
    apiKey: sk-YOUR-API-KEY
    roles:
      - autocomplete
```
